contracts/contracts/VerificationGateway.sol

```diff
@@ -340,6 +340,30 @@ contract VerificationGateway
         return IWallet(blsWallet);
     }
 
+    /**
+     * Utility for measuring the gas used by an operation, suitable for use in
+     * the operation's required gas parameter.
+     *
+     * This has two important differences over standard estimateGas methods:
+     *   1. It does not include gas for calldata, which has already been paid
+     *   2. It works for wallets which don't yet exist
+     */
+    function measureOperationGas(
+        uint256[BLS_KEY_LEN] memory publicKey,
+        IWallet.Operation calldata op
+    ) external returns (uint) {
+        // Don't allow this to actually be executed on chain. Static calls only.
+        require(msg.sender == address(0));
```

Comment:
Prefer `requires` with human readable error messages.
```suggestion
        require(msg.sender == address(0), "VG: read only");
```

Reply:
Will try but we're right up against the 30m gas limit when deploying `VerificationGateway`.